tests/components/rest/test_switch.py

```diff
@@ -16,254 +16,239 @@
-        aioclient_mock.get(self.resource, text="unknown status")
-        asyncio.run_coroutine_threadsafe(
-            self.switch.async_update(), self.hass.loop
-        ).result()
-
-        assert self.switch.is_on is None
-
-    def test_update_timeout(self, aioclient_mock):
-        """Test update when timeout occurs."""
-        aioclient_mock.get(self.resource, exc=asyncio.TimeoutError())
-        asyncio.run_coroutine_threadsafe(
-            self.switch.async_update(), self.hass.loop
-        ).result()
-
-        assert self.switch.is_on is None
+    assert aioclient_mock.call_count == 1
+
+
+async def test_setup(hass, aioclient_mock):
+    """Test setup with valid configuration."""
+    aioclient_mock.get("http://localhost", status=HTTP_OK)
+    assert await async_setup_component(
+        hass,
+        SWITCH_DOMAIN,
+        {
+            SWITCH_DOMAIN: {
+                CONF_PLATFORM: rest.DOMAIN,
+                CONF_NAME: "foo",
+                CONF_RESOURCE: "http://localhost",
+                CONF_HEADERS: {"Content-type": CONTENT_TYPE_JSON},
+                rest.CONF_BODY_ON: "custom on text",
+                rest.CONF_BODY_OFF: "custom off text",
+            }
+        },
+    )
+    assert aioclient_mock.call_count == 1
+    assert_setup_component(1, SWITCH_DOMAIN)
+
+
+async def test_setup_with_state_resource(hass, aioclient_mock):
+    """Test setup with valid configuration."""
+    aioclient_mock.get("http://localhost", status=HTTP_NOT_FOUND)
+    aioclient_mock.get("http://localhost/state", status=HTTP_OK)
+    assert await async_setup_component(
+        hass,
+        SWITCH_DOMAIN,
+        {
+            SWITCH_DOMAIN: {
+                CONF_PLATFORM: rest.DOMAIN,
+                CONF_NAME: "foo",
+                CONF_RESOURCE: "http://localhost",
+                rest.CONF_STATE_RESOURCE: "http://localhost/state",
+                CONF_HEADERS: {"Content-type": CONTENT_TYPE_JSON},
+                rest.CONF_BODY_ON: "custom on text",
+                rest.CONF_BODY_OFF: "custom off text",
+            }
+        },
+    )
+    assert aioclient_mock.call_count == 1
+    assert_setup_component(1, SWITCH_DOMAIN)
+
+
+"""Tests for REST switch platform."""
+
+
+def _setup_test_switch(hass):
+    body_on = Template("on", hass)
+    body_off = Template("off", hass)
+    switch = rest.RestSwitch(
+        NAME,
+        RESOURCE,
+        STATE_RESOURCE,
+        METHOD,
+        HEADERS,
+        AUTH,
+        body_on,
+        body_off,
+        None,
+        10,
+        True,
+    )
+    switch.hass = hass
+    return switch, body_on, body_off
+
+
+def test_name(hass):
+    """Test the name."""
+    switch, body_on, body_off = _setup_test_switch(hass)
+    assert NAME == switch.name
+
+
+def test_is_on_before_update(hass):
+    """Test is_on in initial state."""
+    switch, body_on, body_off = _setup_test_switch(hass)
+    assert switch.is_on is None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)